src/main/java/com/microsoft/azure/relay/AsyncSemaphore.java

```diff
@@ -89,6 +91,10 @@ int availablePermits() {
 		});
 	}
 	
```

Comment:
Let's add some documentation for this function

 /**
  * Acquire the lock without blocking, then execute the given code, 
  * and finally releases the lock regardless if the given supplier threw during its execution
  * @param timeout Maximum time to wait for the lock. Throws TimeoutException if the lock is not acquired after the timeout finishes
  * @param executor The executor that executes the given supplier code
  * @param supplier Code to be executed that returns a CompletionStage after obtining the lock
  * @return A CompletableFuture that completes when the lock is released after returning the desired CompletableFuture from the supplier
  */

Reply:
Good suggestion and I really appreciate your providing it :)

---
In reply to: [259079315](https://github.com/Azure/azure-relay-java/pull/20#discussion_r259079315) [](ancestors = 259079315)